src/charm.py

```diff
@@ -136,6 +138,18 @@ def __init__(self, *args):
             charm=self, relation_name="auth-devices-keys"
         )
 
+        self.blackbox_probes_provider = BlackboxProbesProvider(
+            charm=self,
+            probes=self.self_probe + self.devices_ip_endpoints_probes,
+            refresh_event=[
+                self.on.update_status,
+                self.ingress.on.ready,
+                self.on.config_changed,
+            ],
+        )
+
+        self.framework.observe(self.blackbox_probes_provider.on.invalid_probe, self._on_invalid_probe)
```

Comment:
We're slowly moving away from custom events. Instead of emitting, observing and then setting status out of context, we prefer centralized status setting, by equipping libs with a `collect_status()` method. [Example](https://github.com/canonical/observability-libs/blob/9aa3b5aaf9963e2b0a8c8a431dc04cc59020076e/lib/charms/observability_libs/v0/kubernetes_compute_resources_patch.py#L712).

I think the new blackbox lib should follow this pattern.